themes/default/content/blog/cloud-systems-part-two/index.md

```diff
@@ -0,0 +1,220 @@
+                <li><a href="http://twitter.com/Dixie3Flatline" target="_blank"><i class="fab fa-twitter"></i></a></li>
+                <li><a href="http://linkedin.com/in/katcosgrove" target="_blank"><i class="fab fa-linkedin-in"></i></a></li>
+            </ul>
+    </header>
+{% block content %}{% endblock %}
+</body>
+<script src="https://kit.fontawesome.com/b4747495ea.js" crossorigin="anonymous"></script>
+</html>
+```
+
+This file includes some metadata, stylesheets, the header, navigation, and any Javascript we want to link in. All of the content in this file will exist on every page. Below the header, in the body, you'll see `{% block content %} {% endblock %}`. That's where additional, page-specific content will be added.
+
+`index.html`
+
+```html
+{% extends 'base.html' %}
+
+{% block content %}
+
+<div class="content">
+    <h1>Your Name</h1>
+    <h3>Job Title at Company</h3>
+</div>
+
+{% endblock %}
+```
+
+This is all that exists in our `index.html` now. The first directive tells Jinja2 that this bit of HTML goes in the `base.html` template, and the remaining two wrap the content we want to insert. The same thing is happening in `about.html` and `portfolio.html`. Neat, right? Jinja2 is not unique to Flask, so you can use this templating engine with another web framework if you prefer.
+
+While we could deploy this as-is, it's a bit unwieldy. Let's wrap it up into a container.
+
+## Dockerizing our Website
+
+In the `website` directory, we have something called a Dockerfile. Dockerfiles tell the Docker engine what to do with your application, how your container should behave with respect to the wider internet or other containers, and what needs to be done inside of the container for your application to run. Ours looks like this:
+
+```bash
+FROM ubuntu:20.04
+
+COPY app /app
+
+WORKDIR /app
+
+EXPOSE 80
+
+RUN apt-get update && \
+    apt install -y gcc python3-dev python3-pip python-markupsafe
+
+COPY requirements.txt /app
+
+RUN pip3 install -r requirements.txt
+
+ENTRYPOINT [ "python3" ]
+
+CMD [ "server.py" ]
+```
+
+This Dockerfile says that our container should be running ubuntu:20.04 as its operating system. It will pull the Ubuntu image from Docker Hub, their public image registry, though you can pull from another registry here as well. The `COPY` directive moves our website’s files (the `app` directory) into the container. We then set our working directory for all future commands run inside of the container to `/app` to reduce the amount of typing we have to do later on, update Ubuntu, and install some dependencies using both apt and from the requirements.txt file we also copy over. Lastly, an entrypoint and command to actually start our application are defined. There is a slightly less verbose way to start the application, but this way of defining an `ENTRYPOINT` and `CMD` is fairly common, so it’s worth seeing here.
```

Comment:
```suggestion
This Dockerfile says that our container should be running Ubuntu Linux version 20.04 as its operating system. It will pull the Ubuntu image from Docker Hub, their public image registry, though you can pull from another registry here as well. The `COPY` directive moves our website’s files (the `app` directory) into the container. We then set our working directory for all future commands run inside of the container to `/app` to reduce the amount of typing we have to do later on, update Ubuntu, and install some dependencies using both apt and from the `requirements.txt` file we also copy over. Lastly, an entrypoint and command to actually start our application are defined. There is a slightly less verbose way to start the application, but this way of defining an `ENTRYPOINT` and `CMD` is fairly common, so it’s worth having here.
```